core/src/main/java/com/google/instrumentation/stats/StatsContextFactory.java

```diff
@@ -35,4 +40,69 @@
    * Returns the default {@link StatsContext}.
    */
   public abstract StatsContext getDefault();
+
+  /**
+   * Get current StatsContext from current gRPC Context.
+   *
+   * @return the current {@code StatsContext} from {@code io.grpc.Context}, or {@code null} if
+   *     there's no {@code StatsContext} associated with current {@code io.grpc.Context}.
+   */
+  @Nullable
+  public final StatsContext getCurrentStatsContext() {
+    // TODO(songya): return a no-op StatsContext (or default) when getCurrentStatsContext() returns
```

Comment:
Yeah initially I implemented this method to return default `StatsContext` when `ContextUtils.getCurrentStatsContext()` returns null, because I think if this method returned null, user need to call `getDefault()` again to get a non-null `StatsContext`. I'll update this.